tests/e2e/linalg_ext_ops/top-k-scf-lowering.mlir

```diff
@@ -0,0 +1,193 @@
+  %out_values_empty = tensor.empty() : tensor<1x32xf32>
+  %out_indices_empty = tensor.empty() : tensor<1x32xi32>
+  %neg_inf = arith.constant 0xFF800000 : f32
+  %c0 = arith.constant 0 : i32
+  %out_values = linalg.fill ins(%neg_inf : f32) outs(%out_values_empty : tensor<1x32xf32>) -> tensor<1x32xf32>
+  %out_indices = linalg.fill ins(%c0 : i32) outs(%out_indices_empty : tensor<1x32xi32>) -> tensor<1x32xi32>
+  %0:2 = iree_linalg_ext.topk
+        dimension(1)
+        ins(%input_values: tensor<1x32xf32>)
+        outs(%out_values, %out_indices : tensor<1x32xf32>, tensor<1x32xi32>) {
+        ^bb0(%arg0 : f32, %arg1 : f32):
+         %0 = arith.cmpf ogt, %arg0, %arg1 : f32
+         iree_linalg_ext.yield %0 : i1
+        } -> tensor<1x32xf32>, tensor<1x32xi32>
+
+  check.expect_almost_eq_const(
+      %0#0,
+      dense<[[4.8, 4.23, 4.2, 4.0, 4.0, 4.0, 4.0, 4.0, 3.0, 3.0, 3.0, 3.0, 3.0, 3.0, 3.0, 3.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 1.4, 1.3, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]]> : tensor<1x32xf32>
+  ) : tensor<1x32xf32>
+
+  check.expect_eq_const(
+      %0#1,
+      dense<[[15, 19, 23, 3, 7, 11, 27, 31, 2, 6, 10, 14, 18, 22, 26, 30, 1, 5, 9, 13, 17, 21, 25, 29, 8, 0, 4, 12, 16, 20, 24, 28]]> : tensor<1x32xi32>
+  ) : tensor<1x32xi32>
+
+  return
+}
+
+func.func @vector_topk_test_35() {
+  %input_values = util.unfoldable_constant dense<[[1.3, 2.0, 3.0, 4.0, 1.0, 2.0, 3.0, 4.0, 1.4, 2.0, 3.0, 4.0,
+    1.0, 2.0, 3.0, 4.8, 1.0, 2.0, 3.0, 4.23, 1.0, 2.0, 3.0, 4.2, 1.0, 2.0, 3.0, 4.0, 1.0, 2.0, 3.0, 4.0, 8.0, 8.1, 8.2]]> : tensor<1x35xf32>
+  %out_values_empty = tensor.empty() : tensor<1x35xf32>
+  %out_indices_empty = tensor.empty() : tensor<1x35xi32>
+  %neg_inf = arith.constant 0xFF800000 : f32
+  %c0 = arith.constant 0 : i32
+  %out_values = linalg.fill ins(%neg_inf : f32) outs(%out_values_empty : tensor<1x35xf32>) -> tensor<1x35xf32>
+  %out_indices = linalg.fill ins(%c0 : i32) outs(%out_indices_empty : tensor<1x35xi32>) -> tensor<1x35xi32>
+  %0:2 = iree_linalg_ext.topk
+        dimension(1)
+        ins(%input_values: tensor<1x35xf32>)
+        outs(%out_values, %out_indices : tensor<1x35xf32>, tensor<1x35xi32>) {
+        ^bb0(%arg0 : f32, %arg1 : f32):
+         %0 = arith.cmpf ogt, %arg0, %arg1 : f32
+         iree_linalg_ext.yield %0 : i1
+        } -> tensor<1x35xf32>, tensor<1x35xi32>
+
+  check.expect_almost_eq_const(
+      %0#0,
+      dense<[[8.2, 8.1, 8.0, 4.8, 4.23, 4.2, 4.0, 4.0, 4.0, 4.0, 4.0, 3.0, 3.0, 3.0, 3.0, 3.0, 3.0, 3.0, 3.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 1.4, 1.3, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]]> : tensor<1x35xf32>
+  ) : tensor<1x35xf32>
+
+  check.expect_eq_const(
+      %0#1,
+      dense<[[34, 33, 32, 15, 19, 23, 3, 7, 11, 27, 31, 2, 6, 10, 14, 18, 22, 26, 30, 1, 5, 9, 13, 17, 21, 25, 29, 8, 0, 4, 12, 16, 20, 24, 28]]> : tensor<1x35xi32>
+  ) : tensor<1x35xi32>
+
+  return
+}
+
+// int32 type not supported in the scalar implementation.
```

Comment:
Can we remove them from the mlir file? Are we going to support that? If so, you can add the TODO to `KernelDispatch.cpp` instead of commenting them out.